readthedocs/core/static-src/core/js/readthedocs-doc-embed.js

```diff
@@ -6,11 +6,36 @@ const sphinx = require('./doc-embed/sphinx');
 const search = require('./doc-embed/search');
 const { domReady } = require('./doc-embed/utils');
 
+/*
+ * Inject JQuery if isn't present already.
+ *
+ * Parts of this script rely on JQuery (mainly the flyout menu injection),
+ * since Sphinx no longer includes it, and other tools may not include it,
+ * we must inject it if isn't found before executing our script.
+*/
+function injectJQuery(init) {
+    if (window.jQuery) {
+        init()
+        return
+    }
+    console.log("JQuery not found. Injecting.");
+    var script = document.createElement("script");
+    script.type = "text/javascript";
+    script.src = "https://code.jquery.com/jquery-3.6.3.min.js";
```

Comment:
Should this also be loaded from a readthedocs.io domain because of CSP?

Reply:
Yeah agreed with @benjaoming, for CSP and CORS related headaches, this should be from our own domain.

I'd probably say we should just inject our existing jQuery vendored bundle, which is compiled using Gulp/Bower. Our current vendored jQuery is 2.0 however, but the API hasn't changed too much as well: https://github.com/readthedocs/readthedocs.org/blob/main/bower.json#L17

Reply:
this is loaded from docs domains, we don't have a CSP there, but +1 on serving this from `/_/static/`. About the version, guess, anything that works with https://github.com/readthedocs/sphinx_rtd_theme/blob/0.3.1/js/theme.js should be fine. But since our theme was depending on the jquery version from sphinx not sure if there would be any problems, but running locally should be enough to test it I guess

Reply:
If we look, we can probably find security-related issues addressed in jQuery updates, so it's probably best to stick to the latest version. That is the approach of `sphinxcontrib-jquery` as well, so thus the two approaches can align with the same jQuery version.

For `sphinxcontrib-jquery`, I think the approach has also been that if someone steps up and wants configuration of the jQuery version, it will be implemented... but on-demand. See also: https://github.com/sphinx-contrib/jquery/blob/master/sphinxcontrib/jquery/__init__.py

Reply:
The jQuery version from Sphinx was just which-ever latest version that happened to be vendored when that Sphinx version was released: https://github.com/sphinx-doc/sphinx/commits/v5.3.0/sphinx/themes/basic/static/jquery.js

Reply:
I'm guessing we're also probably okay upgrading the jQuery version installed via Bower, we don't do anything particularly interesting in the existing templates/JS.

Reply:
```
Unable to find a suitable version for jquery, please choose one by typing one of the numbers below:
    1) jquery#~1.8.1 which resolved to 1.8.3 and is required by jquery-ui#1.8.23
    2) jquery#3.6.3 which resolved to 3.6.3 and is required by readthedocs
    3) jquery#>=1.5 which resolved to 3.6.3 and is required by jquery.payment#1.3.3
```

Don't think we are using the other plugins. I went ahead and wrote `2!`, but I don't see the new files being copied to media, hmm, do we have a script for copying the files? :thinking:

Reply:
Traveled back in time to find `npm run vendor`

https://github.com/readthedocs/readthedocs.org/blob/3.11.0/docs/development/front-end.rst

Reply:
I don't recall what we even use jquery-ui for, but the latest version of jquery-ui is a minor version bump and supports jQuery =< 3.6.

Good question on the static files. Vendored libraries are processed with Gulp here:
https://github.com/readthedocs/readthedocs.org/blob/main/gulpfile.js#L193

And are output here:
https://github.com/readthedocs/readthedocs.org/tree/main/readthedocs/static/vendor

Reply:
Ah missed your reply until reload. I thought `gulp build` built all assets, but the command is separate, `gulp vendor`.